frontend/javascript/index.js

```diff
@@ -93,8 +93,20 @@ document.addEventListener("DOMContentLoaded", function(event) {
   });
 });
 
+function contextSave() {
```

Comment:
please rename this to something that provides more context on what it's doing (no pun intended). I would recommend naming it something like `saveAndRestoreNavigationPosition`